app/src/main/java/org/greenstand/android/TreeTracker/permissions/PermissionViewModel.kt

```diff
@@ -0,0 +1,26 @@
+package org.greenstand.android.TreeTracker.permission
+
+import android.location.LocationManager
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+data class PermissionItemsState(
+    val openAppSettings: Boolean = true,
+    val isLocationEnabled: Boolean? = null
+)
+class PermissionViewModel (
+    private val locationManager: LocationManager) : ViewModel(){
+
+    private val _state = MutableLiveData(PermissionItemsState(isLocationEnabled = isLocationEnabled()))
+    val state: LiveData<PermissionItemsState> = _state
+
+    fun isLocationEnabled(): Boolean {
+        return locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)
+    }
```

Comment:
The view should not call method on the viewmodel to get info, it should only rely on the state object. In this case, when the viewmodel is initialized, it can query the GPS status and update the state with the latest value. I'm assuming this was why you have the "isLocationEnabled" in the state object.

Looking at line 107 and 123 I can see the same logic done in two ways. One using state, the other using the viewmodel.